middleware/http_input.go

```diff
@@ -122,7 +122,7 @@ func (mw *HttpInputMiddleware) injectFieldFromUrlAndMap(ptr interface{}) error {
 		}
 
 		src, name := getSourceWayAndName(elType.Field(i))
-		if src == "" && mw.opt.IsReadFromBody {
```

Comment:
We should keep validation `src == ""`, because when use specified source, we do not need to read it from body.